donkeycarManagerFront/src/js/utils.js

```diff
@@ -0,0 +1,38 @@
+// why are we using 2 wide identation again ?
+
+export function getJobDuration(job) {
+  try {
+    const params = JSON.parse(job.parameters);
+    if (typeof job.next_job_details === 'string') {
+      const next_job = JSON.parse(job.next_job_details);
+      return parseInt(params.drive_time) + getJobDuration(next_job);
+    } else {
+      return parseInt(params.drive_time);
+    }
+  } catch(_) {
+    return 0;
```

Comment:
A bit brutal ^^ maybe some errors that can be catched here would be interesting.

It would be better to check if the job as the attribute drive_time (using hasOwnProperty) use it if present, if not log a debug that tells that the job as no duration and that 0s will be used with the job id.

Reply:
I can definitely add a warning on error but I think I'll keep the try catch to prevent broken (potentially nested) jobs breaking everything (i.e next job with no/bad attributes preventing parent job duration from being computed by returning anything but a number)